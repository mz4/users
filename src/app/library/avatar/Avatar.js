import styles from './Avatar.module.scss';

export const Avatar = ({ alt, src }) => {
  return <img className={styles.imageProfile} alt={alt} key={src} src={src} />;
};
