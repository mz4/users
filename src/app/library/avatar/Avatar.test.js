import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Avatar } from './Avatar';

describe('Render Avatar component', () => {
  const avatar = <Avatar alt="avataralt" src="avatarsrc" />;
  const srcUrl = 'http://localhost/avatarsrc';
  it('Create a snapshot', () => {
    const component = renderer.create(avatar);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Render with props', () => {
    const { getByAltText, getByRole } = render(avatar);
    const img = getByRole('img');
    const elementAlt = getByAltText('avataralt');
    expect(elementAlt).toBeDefined();
    expect(img.src).toBe(srcUrl);
  });
});
