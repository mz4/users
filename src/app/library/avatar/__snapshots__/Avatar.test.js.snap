// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Avatar component Create a snapshot 1`] = `
<img
  alt="avataralt"
  className="imageProfile"
  src="avatarsrc"
/>
`;
